zk_watcher.go

```diff
@@ -382,6 +382,42 @@ func (w *zkWatcher) createAll(node string) error {
 	return nil
 }
 
+// triggerCleanup walks the prefix and deletes any non-ephemeral, empty
+// znodes. It swallows any errors encountered.
+func (w *zkWatcher) triggerCleanup() {
+	w.RLock()
+	defer w.RUnlock()
+
+	w.cleanupTree(w.prefix)
+}
+
+func (w *zkWatcher) cleanupTree(node string) bool {
+	children, stat, err := w.conn.Children(node)
+	if err != nil {
+		return false
+	} else if stat.EphemeralOwner() != 0 {
+		return false
+	}
+
+	empty := true
+	for _, child := range children {
+		if !w.cleanupTree(path.Join(node, child)) {
+			empty = false
+		}
+	}
+
+	if empty {
```

Comment:
is this not racey with parent node creation?

can we just yolo delete the parent nodes, and if they aren't empty it'll fail and we just move on with life?